### ISSUE TITLE
[Time Machine Connect] pam_unix(netatalk:auth): authentication failure

### ISSUE THREAD

**Issue by tw-yshuang** [Netatalk/netatalk#179]
# Config
Host:
 `device`: MBP 14 Pro
 `OS`: macOS Monterey 12.4

Server:
`OS`: Ubuntu 20.04
`linux-kernel`: 5.13.0-48-generic
`version`: netatalk 3.1.12
`partition type`: ext4

## */etc/netatalk/afp.conf*
```
;
; Netatalk 3.x configuration file
;

[Global]
; Global server settings

 mimic model = AirPort
 uam list = uams_randnum.so uams_dhx.so uams_dhx2.so

[TimeMachine]
 path = /mnt/TimeMachine
 cnid scheme = dbd
 advertise ssh = yes
 valid users = tmuser
 time machine = yes
```

I already set a user on Ubuntu, e.g. `tmuser:tmuser` that is the same as other config files, created an `ext4` partition, and mounted it on `/mnt/TimeMachine`.
```
drwxr-xr-x 3 tmuser tmuser 4.0K Jun 11 21:29 TimeMachine
```

# Issue
After that, I open Finder on my MBP to connect the server, e.q. `afp://xxx.xxx.xxx.xxx:548`, and it is successfully to connect it, but when I use `Time Machine` to backup, it always fails.

type `journalctl -u netatalk.service` and I found this:
```
pam_unix(netatalk:auth): authentication failure; logname= uid=0 euid=0 tty=afpd ruser=tmuser rhost=::ffff:xxx.xxx.xxx.xxx  user=tmuser
```

I have no idea how to make it work🥲, but it shows my authentication failure, and I am very curious about this.

**Comment by Smiril** [comment]
hello try to use this config

```

;
; Netatalk 3.x configuration file
;

[Global]
hostname = tm

[Homes]
basedir regex = /home

; [My AFP Volume]
; path = /path/to/volume

[Time Machine Volume]
path = /data/tmb
time machine = yes
valid-users = <valid ubuntu user>
vol size limit = 330000
```

and sudo chown <valid ubuntu user> /data/tmb
for have access ... make /data/tmb a btrfs own partition

**Comment by rdmark** [comment]
@tw-yshuang Can you please comment on whether the proposed solution worked for you?

**Comment by tw-yshuang** [comment]
Hi, I already used Samba to achieve TimeMachine, but I will use this method to try it when I have free time(maybe at the end of the month).
Thanks for your reply.

**Comment by rdmark** [comment]
@tw-yshuang Absolutely. Samba is a good choice for modern Macs. Just FWIW, I tried setting up very similar configuration, with an ext4 file system, owned by the staff group, and this afp.conf section for TM:

```
[My Time Machine Volume]
path = /var/local/afpbackup
time machine = yes
valid users = @staff
advertise ssh = yes
```

I've been able to back up hourly for a good long while now from a macOS Ventura system. You original issue most likely had something to do with the user or file system permissions...
